frontend/src/components/MeetingRoom/AudioInputOutputDevices/InputDevices/InputDevices.tsx

```diff
@@ -21,35 +21,23 @@ export type InputDevicesProps = {
  * @returns {ReactElement} - The InputDevices component.
  */
 const InputDevices = ({ handleToggle, customLightBlueColor }: InputDevicesProps): ReactElement => {
-  const { publisher, isPublishing } = usePublisherContext();
-  const [devicesAvailable, setDevicesAvailable] = useState<Device[] | null>(null);
-  const { allMediaDevices } = useDevices();
-  const [options, setOptions] = useState<string[]>([]);
+  const { publisher } = usePublisherContext();
+  const {
+    allMediaDevices: { audioInputDevices },
+  } = useDevices();
 
-  useEffect(() => {
-    setDevicesAvailable(allMediaDevices.audioInputDevices);
-  }, [publisher, allMediaDevices, devicesAvailable, isPublishing]);
-  const changeAudioSource = (deviceId: string) => {
-    publisher?.setAudioSource(deviceId);
-  };
-
-  useEffect(() => {
-    if (devicesAvailable) {
-      const audioDevicesAvailable: string[] = devicesAvailable.map((availableDevice: Device) => {
-        return availableDevice.label;
-      });
-      setOptions(audioDevicesAvailable);
-    }
-  }, [devicesAvailable]);
+  const options = audioInputDevices.map((availableDevice: Device) => {
+    return availableDevice.label;
+  });
 
```

Comment:
These affects and state were not needed and caused multiple renders and chained state effects.
The state comes directly from useDevices, and the state we need can be derived from that state directly, without setting state again. This is how the react docs tells us to do things.